### PR TITLE
fix retry on offline node

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #28072

/kind bug

1. for datanode/querynode/indexnode, if session doesn't exist, it's useless to reset connection and retry.
2. for delegator, stop retry to apply delete when got `NodeNotFound` error

### CODE REVIEW

File: internal/querynodev2/delegator/delegator_data.go [hunk 1]
```diff
@@ -262,7 +262,10 @@ func (sd *shardDelegator) applyDelete(ctx context.Context, nodeID int64, worker
 					PrimaryKeys:  storage.ParsePrimaryKeys2IDs(delRecord.PrimaryKeys),
 					Timestamps:   delRecord.Timestamps,
 				})
-				if errors.Is(err, merr.ErrSegmentNotFound) {
+				if errors.Is(err, merr.ErrNodeNotFound) {
+					log.Warn("try to delete data on non-exist node")
+					return nil
```
Comment (congqixia) [inline internal/querynodev2/delegator/delegator_data.go:267]: maybe return unrecoverable error and mark segment offline here

**Comment by jaime0815** [comment]
rerun ut

**Comment by weiliu1031** [comment]
rerun ut

**Pull request merged**